### PR TITLE
Map single feature selection

### PR DESCRIPTION
**Pull request by Tammo-Feldmann**
## Adds a new selection mode for single selection

In addition to the draw selection on the map this feature offers the user the ability to select individual features one at a time by clicking on the feature. 
Screenshot 2024-05-23 at 3 11 14 PM

### CODE REVIEW

**Review by alukach — COMMENTED** [review]
Really great work, @Tammo-Feldmann .  I know that I'm not a selected viewer on this PR but I was notified none the less so I took a look at the feature.

A few issues stood out:

Once a user enters the select mode, they cannot leave (Hotel California style):

https://github.com/developmentseed/tecnico-energy-app/assets/897290/905c1ae9-89b2-4f87-8dd4-d2f4efe92a5e

While in selection mode, switching scenarios breaks the map color legend and the summary metrics:

https://github.com/developmentseed/tecnico-energy-app/assets/897290/7beda843-a5e7-4de4-80ec-b2fe466ce0d5

**Comment by Tammo-Feldmann** [comment]
@alukach thank you for taking a look. And thanks for finding those issue. I'll make sure to fix both. You're always welcome to be a reviewer, just thought you were too busy.

**Comment by Tammo-Feldmann** [comment]
@alukach the hotel California issue is really just a UI question I believe. You should be able to get out of select mode by clicking on the draw button and vice versa.

**Comment by alukach** [comment]
> the hotel California issue is really just a UI question I believe. You should be able to get out of select mode by clicking on the draw button and vice versa.

In the map view, it seems like there are three interactive states: move (default, you are neither selecting nor drawing), select, and draw.  When you enter the "select" state, how do you get back to the "move" state (ie everything is selected)?  A user could go from "select" to "draw", then draw something, and then hit the "Clear" button, but I don't think that's an intuitive workflow.

**Comment by Tammo-Feldmann** [comment]
> how do you get back to the "move"
Thank you for clarifying. I misunderstood what you were looking for. I pushed a fix for the hotel California issue. I'll leave the other issue for Monday probably since it involves a bit more.

File: components/map/draw-control-pane.tsx
Comment (alukach) [inline components/map/draw-control-pane.tsx:1]: This did the trick! Thanks

**Comment by Tammo-Feldmann** [comment]
@alukach I made an update to how the single selection state works when switching scenarios (or other map state params). I believe this should resolve the issue that you reported above. May I ask you to re-review when you have time?

**Comment by Tammo-Feldmann** [comment]
Note: I had added footprint outlines to the map for the single selection feature since it was hard to see what could be selected when nothing was selected. 
Screenshot 2024-06-03 at 11 46 02 AM

**Review by alukach — APPROVED** [review]
Looks great

**Pull request merged**